code/development/getci.py

```diff
@@ -0,0 +1,134 @@
+img_con = nib.load(file_name_con)
+img_scz = nib.load(file_name_scz)
+data_con = img_con.get_data()[..., 5:]
+data_scz = img_scz.get_data()[..., 5:]
+
+def roi_cor (data, roi1,roi2):
+#input: 
+	# roi1 and roi2 are two list of tuples indicating the voxel 
+	# indexes of the ROI1 and ROI2 respectively
+	# data
+#output: 
+	# returns the mean Fisher's z value of all the correlations among voxels in ROI1 and ROI2
+
+	cor_z = np.zeros((len(roi1),len(roi2)))
+
+	store = np.zeros(cor_z.shape) - float('inf')
+
+	for i in range(0,len(roi1)):
+		for j in range(0,len(roi2)):
+
+			if 0 <= j and j < len(roi1) and 0 <= i and i < len(roi2) and store[j, i] !=  -float('inf'):
+				cor_z[i, j] = store[j, i]
+				continue
+
+			data1 = data[roi1[i]]
+			data2 = data[roi2[j]]
+			cor=np.corrcoef(data1,data2)[1,0]
+			if cor >= 1:
+				cor=0.99999
+			cor_z[i,j] = 1/2*(math.log((1+cor)/(1-cor)))  # Q: how to deal with cor=1 
+
+			store[i, j] = cor_z[i, j]
+			if 0 <= j and j < len(roi1) and 0 <= i and i < len(roi2):
+				store[j, i] = cor_z[i, j]
+
+	return np.mean(cor_z)
+
+def network_cor(data,net1,net2):
+#Input:
+	#net1 and net2 are two lists of lists: (1) list of ROIs (2)list of tuples of one ROI
+#Output: 
+	#95% Confidence Interval of the z-values between networks, a tuple 
+	z_values = np.zeros((len(net1),len(net2)))
+
+	store = np.zeros(z_values.shape) - float('inf')
+
+	for i in range(0,len(net1)):
+		for j in range(0,len(net2)):
+
+			if 0 <= j and j < len(net1) and 0 <= i and i < len(net2) and store[j, i] !=  -float('inf'):
+				z_values[i, j] = store[j, i]
+				continue
+
+			z_values[i,j] = roi_cor(data,net1[i],net2[j])
+
+			store[i, j] = z_values[i, j]
+			if 0 <= j and j < len(net1) and 0 <= i and i < len(net2):
+				store[j, i] = z_values[i, j]
+
+			print "found roi " + str((i, j)) + ". its value is " + str(z_values[i, j])
+
+	# CI = stats.norm.interval(0.95, loc=np.mean(z_values), scale=np.std(z_values))
+	return z_values
+
+
+def ci_within (data,dics):
+#Input: 
+	#triple nesting lists
+	#image data
+#Output:
+	# a list of tuples(CIs); within nework
+	"""
+	ci_wit = []
+	for net in dics:
+		ci_wit.append(network_cor(data,net,net))
+	return ci_wit #CI for ("wDMN","wFP","wCER","wCO")
+	"""
+	return [network_cor(data,rois,rois) for rois in dics]
+
+
+def ci_bewteen (data,dics):
+#Input: 
+	#triple nesting lists
+	#image data
+#Output:
+	#a list of tuples(CIs); between network
+
+	#TODO
+
+	ci_bet = []
+	for i in range(0,len(dics)):
+		for j in range(0,len(dics)):
+			ci_bet.append(network_cor(data,dics[i],dics[j]))
+	return ci_bet #CI for ("bDMN-FP","bDMN-CER","bDMN-CO","bFP-CER","bFP-CO","bCER-CO")
+
+
+def dictolist (dic, mm_to_vox, in_brain_mask):
+	networks=dic.keys() #['Default', 'Fronto-Parietal', 'Cerebellar', 'Cingulo-Opercular']
+	list_net=[]
+	for i in networks:
+		ROIs = dic[i].keys() #['LSF', 'RMPF', 'LMPF', 'RS', 'RSF', 'RIP', 'LiT', 'LIP', 'LpH', 'RpH', 'pCin', 'RiT', 'CT']
```

Comment:
You don't have to change this right now (you can change that when we merge from test branch to master branch). It might be better to output a dictionary instead of relying on the order of the keys here. This is because you should not assume that the caller of this method will always pass in the classes of the right order. Also, dictionary actually does not guarantee the ordering when you call dic.keys(). 

But this is of low priority right now. We can refactor this later.